opta/special_formatter.py

```diff
@@ -14,6 +16,7 @@ def get_field(self, field_name: str, args: Any, kwargs: Any) -> Any:
         try:
             val = super().get_field(field_name, args, kwargs)
         except (KeyError, AttributeError):
+            Interpolation.add(field_name)
```

Comment:
I really don't like the use of global state here. Use of globals makes reasoning about the world and logic harder. I would prefer if we bubble up from the hydrate method any issues, or pass down a "missing keys" object from the caller to keep everything within the call stack.

Reply:
Tried working at it from the hydrate as well, passing down the PartialFormatter with aggregated list of incorrect field names, but that way we would either end up throwing error in an iterative fashion for each incorrect key, and aggregating those errors using PartialFormatter wasn't much help because `hydrate` is called at multiple places with different inputs which would end up just giving partial list of incorrect keys.

Reply:
what's the next step here?

what if we just log a warning message here when the key is missing?
Worst case for the user the warning is frequent, but if they want to get rid of it they can fix their opta file. fair enough?

cc: @rabbitfang @nsarupr

Reply:
@RemyDeWolf 
That can be done. And would be just a small change. We already add a debug log for the same.

Reply:
@nsarupr if this is a small change do you want to do it?
I know it's been approved, but if you think we can do better and it's not too much work, we can do that.

Reply:
@RemyDeWolf @rabbitfang @juandiegopalomino @NitinAgg